### ISSUE TITLE
Can't record system audio & mic together?

### ISSUE THREAD

**Issue by looperg4k1** [kyleneideck/BackgroundMusic#291]
Hi I'm trying to record system audio & mic input together, I'm using Catalina with Background Music v0.3.2, pls see the attached aggregate device setup below. I believe my setup is ok but the result is not expected, I have no idea why.

The problem is I can record & hear the mixed audio(mic & system) in QuickTime Audio Recording only if I selected "High quality", but with the same setting I can't hear the system audio channel from playback(only hear the mic) if I selected "Maximum quality", same happened to QuickTime Movie Recording too although I can see the loudness meter was moving up and down.....Please help~
<img width="400" alt="Screenshotbg" src="https://user-images.githubusercontent.com/48321679/80297051-5d737e80-87b2-11ea-9b97-8a4879a98a3a.png">

**Comment by kyleneideck** [comment]
I'm able to reproduce this, but I think it must be an issue with QuickTime Player rather than Background Music. Setting it to maximum quality should just make it save the audio uncompressed. It must be changing the input device for some reason.

If you need uncompressed audio, maybe you could try using a different program to record like ffmpeg.

**Comment by looperg4k1** [comment]
@kyleneideck Thanks so much for your quick response, actually I want to combine the system audio & mic in Skype/Google Meet for education use, I just tested the app with QuickTime first. Also I tried BlackHole from related projects page but unfortunately I got the same result with the aggregate device setup. :(

**Comment by kyleneideck** [comment]
So it's working in QuickTime Player as long as you don't set it to maximum quality, but it's not working in Skype or Google Meet?

I had a similar problem with Discord and it turned out to be that Discord ignores all but the first (or maybe first and second) channel of your input device. If Skype and Google Meet have the same problem, there might not be an easy solution. It looks like it's possible to make an aggregate  device that only includes some of channels of the devices, but I haven't seen a way to do it without writing some code. (https://stackoverflow.com/q/60445512/1091063)

**Comment by looperg4k1** [comment]
> So it's working in QuickTime Player as long as you don't set it to maximum quality, but it's not working in Skype or Google Meet?

Yes exactly! I just tested on different OS(Mojave) and have the same problem, I suspect it won't work with other platforms like zoom etc too... I don't know much about coding tho, wish I knew how to fix especially in this difficult times.
